### PR TITLE
Fix Cmd+O

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #19663

### CODE REVIEW

File: src/project/internal/projectactionscontroller.cpp [hunk 1]
```diff
@@ -175,7 +175,7 @@ Ret ProjectActionsController::openProject(const ProjectFile& file)
 {
     LOGI() << "Try open project: url = " << file.url.toString() << ", displayNameOverride = " << file.displayNameOverride;
 
-    if (file.url.isLocalFile()) {
+    if (file.isNull() || file.url.isLocalFile()) {
```
Comment (RomanPudashkin) [inline src/project/internal/projectactionscontroller.cpp:178]: At first glance, I thought there was a typo in your solution (I was expecting this condition: `if (!file.isNull() && file.url.isLocalFile()`) ... Ideally, the implementation should look like the following:

```
if (file.isNull()) {
    path = askUser();
    openProject(path);
} else if (file.url.isLocalFile()) {
    openProject(file)
}
```

But at least it's worth adding a comment

Comment (cbjeukendrup) [reply src/project/internal/projectactionscontroller.cpp]: Yeah, I agree that's much nicer. Done!

File: src/project/projecttypes.h [hunk 2]
```diff
@@ -217,7 +222,7 @@ struct ProjectFile {
 
     io::path_t path() const
     {
-        assert(url.isLocalFile());
+        assert(url.isEmpty() || url.isLocalFile());
```
Comment (cbjeukendrup) [inline src/project/projecttypes.h:225]: I'll still keep this here, in order not to trigger unexpected asserts

**Review by RomanPudashkin — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
Tested #19663 on Win10, Mac14, LinuxUbuntu 22.04 - **FIXED**

**Pull request merged**